### ISSUE TITLE
[#470] [live-rig] 503: PREFILL response payload >512MB rejected (state blob 827MB) — RpcClient cap vs KV-scaled state + retry reuses desynced socket

### ISSUE THREAD

**Issue by ddvnguyen** [ddvnguyen/hydra_vortex#594]
## Reproduced (2026-08-09, realistic 2K-token prompt on dense-27b-combined, new core 968134e6c)

Request failed 503 after 117s (model swap + prefill completed fine):

```
[08:34:17] prefill_rpc_error Sid=... Retry=0/3
  System.IO.InvalidDataException: RPC payload length out of range: 827005966 bytes
  at Hydra.Shared.RpcClient.ReadPayloadAsync(RpcClient.cs:437)
[08:34:18] prefill_rpc_error Sid=... Retry=1/3
  System.IO.InvalidDataException: RPC payload length out of range: 272728361713580032 bytes   ← garbage (desynced)
```

Engine side: `PREFILL slot=0 done n_past=7352 kv=413009036 logits=993280B total=827005966` — inference fine; the response carries the state blob (827MB = 413MB KV + logits) per protocol spec ('PREFILL: return n_past and state blob inline', specs/rpc-protocol.md 0x42).

## Root cause (two bugs)

1. **C# cap too small**: `ReadPayloadAsync` (RpcClient.cs:437) rejects payloadLen > `512 * 1024 * 1024` (512MB). The PREFILL response payload is the KV state blob which scales with context — 827MB at 7.3K tokens, ~800MB at 60-80K per CLAUDE.md. Small prompts (KV < 512MB) work; larger contexts always fail.

2. **Retry reuses desynced socket**: `RequestAsync` (RpcClient.cs:64-78) calls `DropConnection()` ONLY on `OperationCanceledException`. `InvalidDataException` (framing error) does NOT drop the connection → the retry reads a misaligned stream (garbage 272728361713580032) → fails again → 503 after retries.

## Fix
1. Raise the cap to accommodate KV-scaled state blobs (≥1GB — 2GB to be safe; verify no other opcode relies on the 512MB limit).
2. On `InvalidDataException` (and any framing/EOF error mid-response), `DropConnection()` before retrying — a framing error means the socket is desynced.
3. Unit test: EnginePrefillAsync with a >512MB payloadLen header → succeeds (cap raised) + connection dropped on bad frame.

## Impact
ANY request with prompt+KV state > ~4K tokens on a 2-slot engine hits this (state blob > 512MB). This blocks the realistic multi-model workflow. Note: this is INDEPENDENT of #592 (health handoff) — #592's fix is live and the routing got past it; this failed deeper, in the prefill RPC response read.
